metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvesterImpl.java

```diff
@@ -258,24 +331,38 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
     }
   }
 
-  private static class ArchiveEntryImpl implements ArchiveEntry {
+  private record ArchiveEntryImpl(String relativeFilePath, ByteArrayInputStream entryContent)
+      implements ArchiveEntry {
+
+    @Override
+    public String getHarvestingIdentifier() {
+      return relativeFilePath;
+    }
 
-    final String entryName;
-    final ByteArrayInputStream entryContent;
+    @Override
+    public void writeContent(OutputStream outputStream) throws IOException {
+       IOUtils.copy(entryContent, outputStream);
+    }
 
-    public ArchiveEntryImpl(String entryName, ByteArrayInputStream entryContent) {
-      this.entryName = entryName;
-      this.entryContent = entryContent;
+    @Override
+    public ByteArrayInputStream getContent() {
+      return entryContent;
     }
 
     @Override
-    public String getEntryName() {
-      return entryName;
+    public boolean isDeleted() {
+      return false;
     }
 
     @Override
+    @Deprecated
     public ByteArrayInputStream getEntryContent() {
-      return entryContent;
+      return getContent();
+    }
+
+    @Override
+    public Instant getTimeStamp() {
+      return null;
```

Comment:
Also here, it always returns null

Reply:
Similar story. Other protocols (can) maintain a timestamp for a record (i.e. last change time). In a zip file we don't have this.